elyra/pipeline/component_registry.py

```diff
@@ -176,52 +173,67 @@ def _read_component_registry(self) -> Tuple[List, List]:
+                    component = self._parser.parse(SimpleNamespace(**component_entry))
+                    if component:
+                        registry['components'].extend(component)
+
+            # Process category entries from catalog
             if 'categories' in catalog_json.keys():
                 for category_id, category_metadata in catalog_json['categories'].items():
 
-                    category_entry = {
-                        "id": category_id,
-                        "label": category_metadata["label"],
-                        "image": category_metadata["image"],
-                        "description": category_metadata["description"]
-                    }
-                    category_entries.append(SimpleNamespace(**category_entry))
+                    category_entry = ComponentCategory(id=category_id,
+                                                       label=category_metadata.get('label'),
+                                                       image_location=category_metadata.get('image'),
+                                                       description=category_metadata.get('description'))
+                    registry['categories'].append(category_entry)
 
-        return component_entries, category_entries
+        return registry
 
-    def _get_relative_location(self, component_type: str, component_path: str):
+    def _get_relative_location(self, component_type: str, component_path: str) -> str:
         """
         Gets the absolute path for a component from a file-based registry
         """
         if component_type == "filename":
             component_path = f"{self._parser._type}/{component_path}"
         return component_path
 
-    def _get_component_registry_entry(self, component_id):
-        """
-        Get the body of the component registry entry with the given id
-        """
-        # Read registry to find list of components
-        component_entries, _ = self._read_component_registry()
-
-        # Find entry with the appropriate id, if exists
-        component_entry = next((entry for entry in component_entries if entry.id == component_id), None)
-        if not component_entry:
-            self.log.error(f"Component with ID '{component_id}' could not be found in the " +
-                           f"{self._component_registry_location} component_id catalog.")
-            raise ValueError(f"Component with ID '{component_id}' could not be found in the " +
-                             f"{self._component_registry_location} component_id catalog.")
-
-        return component_entry
+    def _read_registry_for_component(self, queried_component_id: str, catalog_entry_id: str) -> Component:
+        with open(self._component_registry_location, 'r') as catalog_file:
+            catalog_json = json.load(catalog_file)
+            component_entry = dict(catalog_json['components'].items()).get(catalog_entry_id)
```

Comment:
Isn't `catalog_json['components']` already a dictionary?  This statement seems equivalent to:
```python
component_entry = catalog_json['components'].get(catalog_entry_id)
```

Reply:
Wow 😄 not sure what I was thinking there